File: bindings/lua/Polycode/OpenGLGraphicsInterface.lua
class "OpenGLGraphicsInterface"


function OpenGLGraphicsInterface:__getvar(name)
	if name == "lineSmooth" then
		return Polycode.OpenGLGraphicsInterface_get_lineSmooth(self.__ptr)
	end
end

function OpenGLGraphicsInterface:__setvar(name,value)
	if name == "lineSmooth" then
		Polycode.OpenGLGraphicsInterface_set_lineSmooth(self.__ptr, value)
		return true
	end
	return false
end
function OpenGLGraphicsInterface:OpenGLGraphicsInterface(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.OpenGLGraphicsInterface(unpack(arg))
	end
end

function OpenGLGraphicsInterface:createTexture(texture)
	local retVal = Polycode.OpenGLGraphicsInterface_createTexture(self.__ptr, texture.__ptr)
end

function OpenGLGraphicsInterface:destroyTexture(texture)
	local retVal = Polycode.OpenGLGraphicsInterface_destroyTexture(self.__ptr, texture.__ptr)
end

function OpenGLGraphicsInterface:setViewport(x, y, width, height)
	local retVal = Polycode.OpenGLGraphicsInterface_setViewport(self.__ptr, x, y, width, height)
end

function OpenGLGraphicsInterface:clearBuffers(clearColor, colorBuffer, depthBuffer, stencilBuffer)
	local retVal = Polycode.OpenGLGraphicsInterface_clearBuffers(self.__ptr, clearColor.__ptr, colorBuffer, depthBuffer, stencilBuffer)
end

function OpenGLGraphicsInterface:setParamInShader(shader, param, localParam)
	local retVal = Polycode.OpenGLGraphicsInterface_setParamInShader(self.__ptr, shader.__ptr, param.__ptr, localParam.__ptr)
end

function OpenGLGraphicsInterface:setAttributeInShader(shader, attribute, attributeBinding)
	local retVal = Polycode.OpenGLGraphicsInterface_setAttributeInShader(self.__ptr, shader.__ptr, attribute.__ptr, attributeBinding.__ptr)
end

function OpenGLGraphicsInterface:disableAttribute(shader, attribute)
	local retVal = Polycode.OpenGLGraphicsInterface_disableAttribute(self.__ptr, shader.__ptr, attribute.__ptr)
end

function OpenGLGraphicsInterface:useShader(shader)
	local retVal = Polycode.OpenGLGraphicsInterface_useShader(self.__ptr, shader.__ptr)
end

function OpenGLGraphicsInterface:createProgram(program)
	local retVal = Polycode.OpenGLGraphicsInterface_createProgram(self.__ptr, program.__ptr)
end

function OpenGLGraphicsInterface:destroyProgram(program)
	local retVal = Polycode.OpenGLGraphicsInterface_destroyProgram(self.__ptr, program.__ptr)
end

function OpenGLGraphicsInterface:createVBOForVertexArray(array)
	local retVal = Polycode.OpenGLGraphicsInterface_createVBOForVertexArray(self.__ptr, array.__ptr)
end

function OpenGLGraphicsInterface:createMesh(mesh)
	local retVal = Polycode.OpenGLGraphicsInterface_createMesh(self.__ptr, mesh.__ptr)
end

function OpenGLGraphicsInterface:destroyMesh(mesh)
	local retVal = Polycode.OpenGLGraphicsInterface_destroyMesh(self.__ptr, mesh.__ptr)
end

function OpenGLGraphicsInterface:createShader(shader)
	local retVal = Polycode.OpenGLGraphicsInterface_createShader(self.__ptr, shader.__ptr)
end

function OpenGLGraphicsInterface:destroyShader(shader)
	local retVal = Polycode.OpenGLGraphicsInterface_destroyShader(self.__ptr, shader.__ptr)
end

function OpenGLGraphicsInterface:beginDrawCall()
	local retVal =  Polycode.OpenGLGraphicsInterface_beginDrawCall(self.__ptr)
end

function OpenGLGraphicsInterface:endDrawCall()
	local retVal =  Polycode.OpenGLGraphicsInterface_endDrawCall(self.__ptr)
end

function OpenGLGraphicsInterface:setBlendingMode(blendingMode)
	local retVal = Polycode.OpenGLGraphicsInterface_setBlendingMode(self.__ptr, blendingMode)
end

function OpenGLGraphicsInterface:createRenderBuffer(renderBuffer)
	local retVal = Polycode.OpenGLGraphicsInterface_createRenderBuffer(self.__ptr, renderBuffer.__ptr)
end

function OpenGLGraphicsInterface:destroyRenderBuffer(renderBuffer)
	local retVal = Polycode.OpenGLGraphicsInterface_destroyRenderBuffer(self.__ptr, renderBuffer.__ptr)
end

function OpenGLGraphicsInterface:bindRenderBuffer(renderBuffer)
	local retVal = Polycode.OpenGLGraphicsInterface_bindRenderBuffer(self.__ptr, renderBuffer.__ptr)
end

function OpenGLGraphicsInterface:createVertexBuffer(dataArray)
	local retVal = Polycode.OpenGLGraphicsInterface_createVertexBuffer(self.__ptr, dataArray.__ptr)
end

function OpenGLGraphicsInterface:createIndexBuffer(dataArray)
	local retVal = Polycode.OpenGLGraphicsInterface_createIndexBuffer(self.__ptr, dataArray.__ptr)
end

function OpenGLGraphicsInterface:destroyBuffer(array)
	local retVal = Polycode.OpenGLGraphicsInterface_destroyBuffer(self.__ptr, array.__ptr)
end

function OpenGLGraphicsInterface:drawIndices(type, indexArray)
	local retVal = Polycode.OpenGLGraphicsInterface_drawIndices(self.__ptr, type, indexArray.__ptr)
end

function OpenGLGraphicsInterface:drawArrays(type, vertexCount)
	local retVal = Polycode.OpenGLGraphicsInterface_drawArrays(self.__ptr, type, vertexCount)
end

function OpenGLGraphicsInterface:enableDepthTest(val)
	local retVal = Polycode.OpenGLGraphicsInterface_enableDepthTest(self.__ptr, val)
end

function OpenGLGraphicsInterface:enableDepthWrite(val)
	local retVal = Polycode.OpenGLGraphicsInterface_enableDepthWrite(self.__ptr, val)
end

function OpenGLGraphicsInterface:enableBackfaceCulling(val)
	local retVal = Polycode.OpenGLGraphicsInterface_enableBackfaceCulling(self.__ptr, val)
end

function OpenGLGraphicsInterface:setLineSize(lineSize)
	local retVal = Polycode.OpenGLGraphicsInterface_setLineSize(self.__ptr, lineSize)
end

function OpenGLGraphicsInterface:setWireframeMode(val)
	local retVal = Polycode.OpenGLGraphicsInterface_setWireframeMode(self.__ptr, val)
end

function OpenGLGraphicsInterface:enableScissor(val)
	local retVal = Polycode.OpenGLGraphicsInterface_enableScissor(self.__ptr, val)
end

function OpenGLGraphicsInterface:setScissorBox(box)
	local retVal = Polycode.OpenGLGraphicsInterface_setScissorBox(self.__ptr, box.__ptr)
end

function OpenGLGraphicsInterface:__delete()
	if self then Polycode.delete_OpenGLGraphicsInterface(self.__ptr) end
end

File: bindings/javascript/Polycode/OpenGLGraphicsInterface.js
function OpenGLGraphicsInterface() {
}

OpenGLGraphicsInterface.prototype.createTexture = function(texture) {
	Polycode.OpenGLGraphicsInterface_createTexture(this.__ptr, texture)
}

OpenGLGraphicsInterface.prototype.destroyTexture = function(texture) {
	Polycode.OpenGLGraphicsInterface_destroyTexture(this.__ptr, texture)
}

OpenGLGraphicsInterface.prototype.setViewport = function(x,y,width,height) {
	Polycode.OpenGLGraphicsInterface_setViewport(this.__ptr, x,y,width,height)
}

OpenGLGraphicsInterface.prototype.clearBuffers = function(clearColor,colorBuffer,depthBuffer,stencilBuffer) {
	Polycode.OpenGLGraphicsInterface_clearBuffers(this.__ptr, clearColor,colorBuffer,depthBuffer,stencilBuffer)
}

OpenGLGraphicsInterface.prototype.setParamInShader = function(shader,param,localParam) {
	Polycode.OpenGLGraphicsInterface_setParamInShader(this.__ptr, shader,param,localParam)
}

OpenGLGraphicsInterface.prototype.setAttributeInShader = function(shader,attribute,attributeBinding) {
	Polycode.OpenGLGraphicsInterface_setAttributeInShader(this.__ptr, shader,attribute,attributeBinding)
}

OpenGLGraphicsInterface.prototype.disableAttribute = function(shader,attribute) {
	Polycode.OpenGLGraphicsInterface_disableAttribute(this.__ptr, shader,attribute)
}

OpenGLGraphicsInterface.prototype.useShader = function(shader) {
	Polycode.OpenGLGraphicsInterface_useShader(this.__ptr, shader)
}

OpenGLGraphicsInterface.prototype.createProgram = function(program) {
	Polycode.OpenGLGraphicsInterface_createProgram(this.__ptr, program)
}

OpenGLGraphicsInterface.prototype.destroyProgram = function(program) {
	Polycode.OpenGLGraphicsInterface_destroyProgram(this.__ptr, program)
}

OpenGLGraphicsInterface.prototype.createVBOForVertexArray = function(array) {
	Polycode.OpenGLGraphicsInterface_createVBOForVertexArray(this.__ptr, array)
}

OpenGLGraphicsInterface.prototype.createMesh = function(mesh) {
	Polycode.OpenGLGraphicsInterface_createMesh(this.__ptr, mesh)
}

OpenGLGraphicsInterface.prototype.destroyMesh = function(mesh) {
	Polycode.OpenGLGraphicsInterface_destroyMesh(this.__ptr, mesh)
}

OpenGLGraphicsInterface.prototype.createShader = function(shader) {
	Polycode.OpenGLGraphicsInterface_createShader(this.__ptr, shader)
}

OpenGLGraphicsInterface.prototype.destroyShader = function(shader) {
	Polycode.OpenGLGraphicsInterface_destroyShader(this.__ptr, shader)
}

OpenGLGraphicsInterface.prototype.beginDrawCall = function() {
	Polycode.OpenGLGraphicsInterface_beginDrawCall(this.__ptr)
}

OpenGLGraphicsInterface.prototype.endDrawCall = function() {
	Polycode.OpenGLGraphicsInterface_endDrawCall(this.__ptr)
}

OpenGLGraphicsInterface.prototype.setBlendingMode = function(blendingMode) {
	Polycode.OpenGLGraphicsInterface_setBlendingMode(this.__ptr, blendingMode)
}

OpenGLGraphicsInterface.prototype.createRenderBuffer = function(renderBuffer) {
	Polycode.OpenGLGraphicsInterface_createRenderBuffer(this.__ptr, renderBuffer)
}

OpenGLGraphicsInterface.prototype.destroyRenderBuffer = function(renderBuffer) {
	Polycode.OpenGLGraphicsInterface_destroyRenderBuffer(this.__ptr, renderBuffer)
}

OpenGLGraphicsInterface.prototype.bindRenderBuffer = function(renderBuffer) {
	Polycode.OpenGLGraphicsInterface_bindRenderBuffer(this.__ptr, renderBuffer)
}

OpenGLGraphicsInterface.prototype.createVertexBuffer = function(dataArray) {
	Polycode.OpenGLGraphicsInterface_createVertexBuffer(this.__ptr, dataArray)
}

OpenGLGraphicsInterface.prototype.createIndexBuffer = function(dataArray) {
	Polycode.OpenGLGraphicsInterface_createIndexBuffer(this.__ptr, dataArray)
}

OpenGLGraphicsInterface.prototype.destroyBuffer = function(array) {
	Polycode.OpenGLGraphicsInterface_destroyBuffer(this.__ptr, array)
}

OpenGLGraphicsInterface.prototype.drawIndices = function(type,indexArray) {
	Polycode.OpenGLGraphicsInterface_drawIndices(this.__ptr, type,indexArray)
}

OpenGLGraphicsInterface.prototype.drawArrays = function(type,vertexCount) {
	Polycode.OpenGLGraphicsInterface_drawArrays(this.__ptr, type,vertexCount)
}

OpenGLGraphicsInterface.prototype.enableDepthTest = function(val) {
	Polycode.OpenGLGraphicsInterface_enableDepthTest(this.__ptr, val)
}

OpenGLGraphicsInterface.prototype.enableDepthWrite = function(val) {
	Polycode.OpenGLGraphicsInterface_enableDepthWrite(this.__ptr, val)
}

OpenGLGraphicsInterface.prototype.enableBackfaceCulling = function(val) {
	Polycode.OpenGLGraphicsInterface_enableBackfaceCulling(this.__ptr, val)
}

OpenGLGraphicsInterface.prototype.setLineSize = function(lineSize) {
	Polycode.OpenGLGraphicsInterface_setLineSize(this.__ptr, lineSize)
}

OpenGLGraphicsInterface.prototype.setWireframeMode = function(val) {
	Polycode.OpenGLGraphicsInterface_setWireframeMode(this.__ptr, val)
}

OpenGLGraphicsInterface.prototype.enableScissor = function(val) {
	Polycode.OpenGLGraphicsInterface_enableScissor(this.__ptr, val)
}

OpenGLGraphicsInterface.prototype.setScissorBox = function(box) {
	Polycode.OpenGLGraphicsInterface_setScissorBox(this.__ptr, box)
}
